### ISSUE TITLE
Closing StringIO not idempotent

### ISSUE THREAD

**Issue by janko** [oracle/truffleruby#1640]
Hello :wave:

The `#close` method in IO and IO-like objects is idempotent in MRI:

```rb
file = File.open("path")
file.close
file.close # does nothing
```
```rb
require "stringio"
io = StringIO.new
io.close
io.close # does nothing
```

In TruffleRuby RC 14, closing File objects is indeed idempotent, but for StringIO objects an `IOError: closed stream` exception is raised on second `#close`.

**Comment by nirvdrum** [comment]
Thanks for the report. I looked into it and this was behavior that changed with Ruby 2.3. Prior to that, MRI would also raise an exception. I'll get this fixed up on our end and add a spec for it as well.

**Comment by nirvdrum** [comment]
As it turns out there, there was already a spec but we had it tagged. This is all fixed up now by c298af81ff037af0e24315b5a3b5f71ebe25496c and will be part of the next release (1.0.0-rc15). Thanks again for filing the issue!

**Comment by janko** [comment]
Great, thanks for the fix!
